### ISSUE TITLE
Bug: URLs with text fragments are improperly detected

### ISSUE THREAD

**Issue by non-bin** [platers/obsidian-linter#1568]
- [x] I have verified that I am on the latest version of the Linter

## Describe the Bug

[Text fragments](https://developer.mozilla.org/en-US/docs/Web/URI/Reference/Fragment/Text_fragments) in URLs are not detected as part of the url

## How to Reproduce

Enable [no-bare-urls](https://platers.github.io/obsidian-linter/settings/content-rules/#no-bare-urls)

Before

``` markdown
https://example.com/page#:~:text=fragment
<https://example.com/page#:~:text=fragment>
https://example.com/page#normalFragment
<https://example.com/page#normalFragment>
```

After

``` markdown
<https://example.com/page>#:~:text=fragment
<<https://example.com/page>#:~:text=fragment>
<https://example.com/page#normalFragment>
<https://example.com/page#normalFragment>
```

## Expected Behaviour

It should include the text fragment

Before

``` markdown
https://example.com/page#:~:text=fragment
<https://example.com/page#:~:text=fragment>
https://example.com/page#normalFragment
<https://example.com/page#normalFragment>
```

After

``` markdown
<https://example.com/page#:~:text=fragment>
<https://example.com/page#:~:text=fragment>
<https://example.com/page#normalFragment>
<https://example.com/page#normalFragment>
```

## Device

- [x] Desktop
- [ ] Mobile

**Comment by pjkaufman** [comment]
A fix for this has been merged. It is not in the release from today, but it should be on master and in the next release. Please let us know if the issue persists in either of those places.
